### PR TITLE
Added 5 Jasmine tests for String.Trim function.

### PR DESCRIPTION
**Pull request by agamble33**
Added four more unit tests for trim method

### CODE REVIEW

File: src/main/javascript/software.bytepushers.string.extensions.js [hunk 1]
```diff
@@ -16,7 +16,7 @@
      * @author <a href="mailto:pouncilt.developer@gmail.com">Tont&eacute; Pouncil</a>
      */
     String.prototype.trim = function () {
-        return this.replace(/^\s+|\s+$/g, '');
+        return this.replace(/^\s+|\s+$/g, '');
```
Comment (tonte-pouncil) [inline src/main/javascript/software.bytepushers.string.extensions.js]: Remove space at the end of line.

File: src/test/javascript/stringExtensionSpec.js [hunk 2]
```diff
@@ -1,46 +1,79 @@
-define(['bytepushers'], function(BytePushers) {
-    describe("String extension tests:", function() {
+define(['bytepushers'], function (BytePushers) {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 3]
```diff
@@ -1,46 +1,79 @@
...
+    describe("String extension tests:", function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 4]
```diff
@@ -1,46 +1,79 @@
...
 
         var testStr = "Hello this is a random string";
 
         describe('String.prototype.trim', function () {
-            it('will trim whitespace correctly', function() {
+            it('will trim whitespace correctly', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 5]
```diff
@@ -1,46 +1,79 @@
...
                 var result = "                       l     shdasdk;lg;al j          ".trim();
 
                 expect(result).toBe("l     shdasdk;lg;al j");
             });
+
+            it('trim white space before and after string but not space between letters', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 6]
```diff
@@ -1,46 +1,79 @@
...
+                var testString = " a b c ";
+                var expectedResult = "a b c";
+                var actualResult = testString.trim();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+            it('trim white space before and after string but not space between words', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 7]
```diff
@@ -1,46 +1,79 @@
...
+                var testString = " Donkey Kong ";
+                var expectedResult = "Donkey Kong";
+                var actualResult = testString.trim();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+            it('trim white space after string but not space between words', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 8]
```diff
@@ -1,46 +1,79 @@
...
+                var testString = "Duck Hunt ";
+                var expectedResult = "Duck Hunt";
+                var actualResult = testString.trim();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+            it('white space will not be trimmed if there is no white space', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 9]
```diff
@@ -1,46 +1,79 @@
...
+                var actualResult = testString.trim();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
         });
 
         describe('String.prototype.includes', function () {
-            it('can tell if a string includes a string', function() {
+            it('can tell if a string includes a string', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 10]
```diff
@@ -1,46 +1,79 @@
...
                 var result = testStr.includes("o this i");
 
                 expect(result).toBe(true);
             });
 
-            it('can tell if a string does not include a string', function() {
+            it('can tell if a string does not include a string', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 11]
```diff
@@ -1,46 +1,79 @@
...
                 var result = testStr.includes("This is not in the string");
 
                 expect(result).toBe(false);
             });
         });
 
         describe('String.prototype.toCamelCase', function () {
-            it('can format a sentence into camel case', function() {
+            it('can format a sentence into camel case', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 12]
```diff
@@ -1,46 +1,79 @@
...
                 var result = "Hello this is a sentence.".toCamelCase();
 
                 expect(result).toBe("helloThisIsASentence.");
             });
         });
 
         describe('String.prototype.toNormalCase', function () {
-            it('can format a sentence into camel case', function() {
+            it('can format a sentence into camel case', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 13]
```diff
@@ -1,46 +1,79 @@
...
                 var result = "Hello world".toCamelCase().toNormalCase();
                 // helloWorld => "Hello World"
                 expect(result).toBe("Hello World");
             });
 
-            it('can format a sentence into camel case with a single letter word', function() {
+            it('can format a sentence into camel case with a single letter word', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 14]
```diff
@@ -51,7 +84,7 @@ define(['bytepushers'], function(BytePushers) {
         });
 
         describe('String.format', function () {
-            it('can format a string properly', function() {
+            it('can format a string properly', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 15]
```diff
@@ -1,46 +1,79 @@
...
         describe('String.prototype.trim', function () {
-            it('will trim whitespace correctly', function() {
+            it('will trim whitespace correctly', function () {
                 var result = "                       l     shdasdk;lg;al j          ".trim();
 
                 expect(result).toBe("l     shdasdk;lg;al j");
             });
+
+            it('trim white space before and after string but not space between letters', function () {
+                var testString = " a b c ";
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Rename testString variable to stringUnderTest.

File: src/test/javascript/stringExtensionSpec.js [hunk 16]
```diff
@@ -1,46 +1,79 @@
...
+                var expectedResult = "a b c";
+                var actualResult = testString.trim();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+            it('trim white space before and after string but not space between words', function () {
+                var testString = " Donkey Kong ";
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Rename testString variable to stringUnderTest.

File: src/test/javascript/stringExtensionSpec.js [hunk 17]
```diff
@@ -1,46 +1,79 @@
...
+                var expectedResult = "Donkey Kong";
+                var actualResult = testString.trim();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+            it('trim white space after string but not space between words', function () {
+                var testString = "Duck Hunt ";
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Rename testString variable to stringUnderTest.

File: src/test/javascript/stringExtensionSpec.js [hunk 18]
```diff
@@ -1,46 +1,79 @@
...
+                var expectedResult = "Duck Hunt";
+                var actualResult = testString.trim();
+
+                expect(actualResult).toBe(expectedResult);
+            });
+
+            it('white space will not be trimmed if there is no white space', function () {
+                var testString = "Ness";
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Rename testString variable to stringUnderTest.

**Review by tonte-pouncil — CHANGES_REQUESTED** [review]

**Review by tonte-pouncil — APPROVED** [review]

**Comment by ocampd1** [comment]
Already merged.

**Pull request closed without merging**